template_agent/utils/pylogger.py

```diff
@@ -1,204 +1,214 @@
-
-
-def get_uvicorn_log_config(log_level: str = "INFO") -> Dict[str, Any]:
-    """Return a Uvicorn-compatible logging config that integrates with structlog."""
+from tqdm import tqdm
+
+from template_agent.utils.constants import LOGGER
+from template_agent.utils.trace_context import get_log_context, get_trace_id
+
+
+def get_log_file_path(default_path: str = "/etc/logs/app.log") -> str:
+    """Get the log file path with directory creation and fallback handling.
+
+    Args:
+        default_path: Default log file path if LOG_FILE_PATH env var is not set
+
+    Returns:
+        Valid log file path that can be written to
+    """
+    import os
+    import tempfile
+
+    log_file_path = os.environ.get("LOG_FILE_PATH", default_path)
+
+    # Ensure log directory exists
+    log_dir = os.path.dirname(log_file_path)
+    if log_dir and not os.path.exists(log_dir):
+        try:
+            os.makedirs(log_dir, exist_ok=True)
+        except (OSError, PermissionError):
+            # Fallback to temp directory if log directory is not writable
+            log_file_path = os.path.join(tempfile.gettempdir(), "app.log")
+
+    return log_file_path
+
+
+def get_default_log_format() -> str:
+    """Get the default log format string used across the application.
+
+    Returns:
+        Standard log format optimized for Splunk ingestion
+    """
+    return "timestamp=%(asctime)s.%(msecs)03d log_level=%(levelname)s hostname=%(hostname)s environment=%(environment)s trace_id=%(trace_id)s client.name=%(client_name)s client.version=%(client_version)s jwt.client_id=%(jwt_client_id)s jwt.username=%(jwt_username)s http.origin=%(http_origin)s http.method=%(http_method)s http.path=%(http_path)s user_agent=%(user_agent)s class=%(module)s function=%(funcName)s log_message=%(message)s"
+
+
+def get_default_log_date_format() -> str:
+    """Get the default date format string used for logging.
+
+    Returns:
+        Standard date format for log timestamps
+    """
+    return "%Y-%m-%d %H:%M:%S"
+
+
+def get_log_rotation_config() -> dict:
+    """Get the standard log rotation configuration.
+
+    Returns:
+        Dictionary with log rotation settings
+    """
+    return {
+        "maxBytes": 52428800,  # 50MB for better Splunk ingestion
+        "backupCount": 5,  # Keep 5 backup files
+        "encoding": "utf-8",
+    }
+
+
+class TqdmLoggingHandler(logging.StreamHandler):
+    """Custom logging handler that uses tqdm.write to avoid interfering with progress bars."""
+
+    def __init__(self, level=logging.NOTSET):
+        """Initialize the TqdmLoggingHandler with the specified logging level."""
+        super().__init__(level)
+
+    def emit(self, record):
+        """Emit a log record using tqdm.write to avoid progress bar interference."""
+        try:
+            msg = self.format(record)
+            tqdm.write(
+                msg
+            )  # Use tqdm's write method to ensure output doesn't interfere with progress bars
```

Comment:
If you put the line comment which here is on line 91 instead on the previous line (i.e., before line 89), then the formatter won't spread line 89 across _three_  lines.  😝 

That aside, do you really need that line comment, given the explanation at line 86?  I recommend omitting it from line 89-91 altogether.